### ISSUE TITLE
`state rm` or `state mv` with remote state (aws s3) broken on 0.10.0

### ISSUE THREAD

**Issue by markwellis** [hashicorp/terraform#15746]
### Terraform Version
Terraform v0.10.0

### Terraform Configuration Files
```hcl

provider "aws" {}                                           
                                                            
terraform {                                                 
    backend "s3" {                                          
        bucket = "tfbucket"              
        key    = "foo/terraform.tfstate"
        region = "eu-west-1"                                
        encrypt = "true"                                    
    }                                                       
} 
```

### Debug Output
2017/08/07 11:19:54 [WARN] Invalid log level: "1". Defaulting to level: TRACE. Valid levels are: [TRACE DEBUG INFO WARN ERROR]
2017/08/07 11:19:54 [INFO] Terraform version: 0.10.0  2041053ee9444fa8175a298093b55a89586a1823
2017/08/07 11:19:54 [INFO] Go runtime version: go1.8
2017/08/07 11:19:54 [INFO] CLI args: []string{"/usr/local/bin/terraform", "state", "rm", "aws_eip.nat[1]"}
2017/08/07 11:19:54 [DEBUG] Attempting to open CLI config file: /home/mark/.terraformrc
2017/08/07 11:19:54 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2017/08/07 11:19:54 [INFO] CLI command args: []string{"state", "rm", "aws_eip.nat[1]"}
2017/08/07 11:19:54 [DEBUG] plugin: waiting for all plugin processes to complete...
No state file was found!

State management commands require a state file. Run this command
in a directory where Terraform has been run or use the -state flag
to point the command to a specific state location.

### Expected Behavior
remote state should be used and item removed

### Actual Behavior
state not found error

### Steps to Reproduce
1. `terraform state rm $thing`
2. `terraform state mv aws_eip.nat[1] aws_eip.nbf`

### Important Factoids
works in 0.9.11
other `state` commands work (list, pull, push, show)

**Comment by thiagocaiubi** [comment]
I'm facing the same issue with azurerm provider. Looks more like a terraform core issue rather than with providers.

**Comment by jbardin** [comment]
Hi @markwellis,

Thanks for filing the issue. This was fixed with #15652, and will be in the next release.

**Comment by rturam** [comment]
Is it deployed in the latest release? I am facing the same issue. Removed the state and after re-run, that state is not found. Am I missing something?
